buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/JUnitXmlTestEventsGenerator.groovy

```diff
@@ -51,41 +54,36 @@ class JUnitXmlTestEventsGenerator {
         this.testListenerBroadcast = testListenerBroadcast
     }
 
-    void processXmlFile(File resultsFile, Object build) {
-        processXml(new XmlSlurper().parse(resultsFile), build)
-    }
-
-    void processXml(GPathResult testResult, Object build) {
-        String suiteName = testResult.@name.text()
-        def testSuiteDescriptor = new DecoratingTestDescriptor(new DefaultTestClassDescriptor(0, suiteName), createWorkerSuite())
+    void processTestSuite(JUnitTestSuite testSuite, Object build) {
+        String suiteName = testSuite.name
+        DecoratingTestDescriptor testSuiteDescriptor = new DecoratingTestDescriptor(new DefaultTestClassDescriptor(0, suiteName), createWorkerSuite())
         testListener.beforeSuite(testSuiteDescriptor.parent.parent)
         testListener.beforeSuite(testSuiteDescriptor.parent)
         testListener.beforeSuite(testSuiteDescriptor)
-        String timestamp = testResult.@timestamp.text()
+        String timestamp = testSuite.timestamp
         long startTime = DatatypeFactory.newInstance().newXMLGregorianCalendar(timestamp).toGregorianCalendar().getTimeInMillis()
-        testResult.testcase.each { testCase ->
-            String testCaseClassName = testCase.@classname.text()
-            String testMethodName = testCase.@name.text()
-            def testCaseDescriptor = new DecoratingTestDescriptor(new DefaultTestMethodDescriptor(0, testCaseClassName, testMethodName), testSuiteDescriptor)
-            def skipped = testCase.skipped
-            def failure = testCase.failure
-            long endTime = startTime + (testCase.@time.toDouble() * 1000).round()
-
-            if (failure.size() > 0) {
+        testSuite.testCases.each { JUnitTestCase testCase ->
+            String testCaseClassName = testCase.className
+            String testMethodName = testCase.name
+            DecoratingTestDescriptor testCaseDescriptor = new DecoratingTestDescriptor(new DefaultTestMethodDescriptor(0, testCaseClassName, testMethodName), testSuiteDescriptor)
+            List<JUnitFailure> failures = testCase.failures
+            long endTime = startTime + (testCase.time * 1000).round()
+
+            if (failures && failures.size() > 0) {
                 testListener.beforeTest(testCaseDescriptor)
                 publishAdditionalMetadata(testCaseDescriptor, build)
                 try {
-                    String systemErr = testResult."system-err".text()
+                    String systemErr = testSuite.systemOut
```

Comment:
Using `systemErr` was intentional. Why do you want to switch this to system out?

Reply:
A typo...sorry. :-(